src/GovUk.Education.ExploreEducationStatistics.Admin/Services/DataBlockService.cs

```diff
@@ -122,9 +122,32 @@ public class DataBlockService : IDataBlockService
                 {
                     var viewModel = _mapper.Map<DataBlockViewModel>(dataBlock);
 
+                    var subjectId = dataBlock.Query.SubjectId;
+
+                    var releaseContentBlock = await _context.ReleaseContentBlocks
+                        .Include(rcb => rcb.Release)
+                        .SingleOrDefaultAsync(rcb => rcb.ContentBlockId == id);
```

Comment:
All data blocks will have a `ReleaseContentBlock` so this should have been using `Single` rather than `SingleOrDefaultAsync`. Currently you'll have a warning that `releaseContentBlock` can be null due to using `SingleOrDefaultAsync` which your ignoring.

See other comment though as fetching the content block and also including all the fields of the release just to get the id of the release can be simplified.